src/contents.ts

```diff
@@ -742,4 +745,17 @@ namespace ContentsManager {
   function extname(path: string): string {
     return posix.extname(path);
   }
+
+  /**
+   * Normalize a file extension to be of the type `'.foo'`.
+   *
+   * Adds a leading dot if not present and converts to lower case.
+   */
+  export
+  function normalizeExtension(extension: string): string {
+    if (extension.indexOf('.') !== 0) {
```

Comment:
`normalizeExtension('')` should still return `''`, so you may also want `extension.length > 0 &&` here.